deptofdefense/move.mil
# Reorganize seed data and scripts

## Checklist

I have…


## Summary of Changes

This pull request refactors the code performing data seeding. It moves code from the `db/seeds.rb` file into unique classes, namespaced under the `Seeds` module. The classes are fairly consistent with some minor deviations in interface for the `BaselineRates` seed class.

I believe this structure provides a bit more organization than what we had while preserving functionality. It's also now possible to seed specific bits of information directly via the console:

```sh
± bin/rails c
Loading development environment (Rails 5.1.4)
irb(main):001:0> require Rails.root.join('db', 'seeds', 'faqs.seeds.rb')
=> true
irb(main):002:0> Seeds::Faqs.new.seed!
```

This PR also moves seed-related data files from `db/seeds` to `lib/data`. `lib/data` seemed like a good logical choice and it tidies up the `db/seeds` folder, but I'm flexible on this if anyone feels strongly one way or the other.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout refactor-seeds`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. run `bin/setup` (if not already run during setup from previous step),
1. run `bin/rails server`,
1. load up http://localhost:3000 and click through various features relying on seed data (e.g. FAQs, Tutorials, Entitlements, Locator Map, PPM Estimator), confirming that features function as intended.
